app/uk/gov/hmrc/helptosavestub/controllers/SquidController.scala

```diff
@@ -0,0 +1,304 @@
+    val errorMap: Map[String, Map[String, String]] =
+      Map("error" ->
+        Map("errorMessageId" -> triple._1,
+          "errorMessage" -> triple._2,
+          "errorDetail" -> triple._3))
+    Json.toJson(errorMap)
+  }
+
+  private def hasNumericChars(str: String): Boolean = """.*\d.*""".r.pattern.matcher(str).matches
+
+  private def hasDisallowedChars(str: String): Boolean = !"""^[a-zA-Z&\.-]*$""".r.pattern.matcher(str).matches
+
+  private def hasSpecialInFirstPlace(str: String): Boolean = """^[&\.-].*""".r.pattern.matcher(str).matches
+
+  private def hasSpecialInLastPlace(str: String): Boolean = """.*[&\.-]$""".r.pattern.matcher(str).matches
+
+  private def hasInsufficientAlphaCharsAtStart(str: String) = !"^[a-zA-Z]{3}.*".r.pattern.matcher(str).matches
+
+  private def hasInsufficientConsecutiveAlphaChars(str: String) = !"^.*[a-zA-Z]{4}.*".r.pattern.matcher(str).matches
+
+  private def hasTooManyConsecutiveSpecialChars(str: String) = """^.*[&\.-]{2}.*""".r.pattern.matcher(str).matches
+
+  private def invalidPostcode(postcode: String): Boolean = {
+    val noSpacesPostcode = postcode.filter {
+      _ != ' '
+    }
+    val row2Expr = """^[A-P|R-U|WYZ]\d\d[AB|D-H|JLN|P-U|W-Z][AB|D-H|JLN|P-U|W-Z]$""".r
+    val row3Expr = """^[A-P|R-U|WYZ]\d\d\d[AB|D-H|JLN|P-U|W-Z][AB|D-H|JLN|P-U|W-Z]$""".r
+    val row4Expr = """^[A-P|R-U|WYZ][A-H|K-Y]\d\d[AB|D-H|JLN|P-U|W-Z][AB|D-H|JLN|P-U|W-Z]$""".r
+    val row5Expr = """^[A-P|R-U|WYZ][A-H|K-Y]\d\d\d[AB|D-H|JLN|P-U|W-Z][AB|D-H|JLN|P-U|W-Z]$""".r
+    val row6Expr = """^[A-P|R-U|WYZ][A-H|K-Y]\d[ABEHMNPR|V-Y]\d[AB|D-H|JLN|P-U|W-Z][AB|D-H|JLN|P-U|W-Z]$""".r
+    val row7Expr = """^[A-P|R-U|WYZ]\d[A-H|HJK|S-U|W]\d[AB|D-H|JLN|P-U|W-Z][AB|D-H|JLN|P-U|W-Z]$""".r
+    val row8Expr = "^[A-Z]1ZZ$".r
+    val row9Expr = "^[A-Z][A-Z]1ZZ$".r
+    val row10Expr = "^[A-Z][A-Z][A-Z]1ZZ$".r
+    val row11Expr = "^[A-Z][A-Z][A-Z][A-Z]1ZZ$".r
+    val row12Expr = """^BFPO\d$""".r
+    val row13Expr = """^BFPO\d\d$""".r
+    val row14Expr = """^BFPO\d\d\d$""".r
+    val row15Expr = """^BFPO\d\d\d\d$""".r
+
+    (noSpacesPostcode != "GIR0AA") &&
+      (noSpacesPostcode match {
+        case row2Expr() | row3Expr() | row4Expr() | row5Expr() | row6Expr() | row7Expr() | row8Expr() | row9Expr()
+             | row10Expr() | row11Expr() | row12Expr() | row13Expr() | row14Expr() | row15Expr() => false
+        case _ => true
+      })
+
+    //        val noSpacesPostcode = postcode.filter{_ != ' '}
+    //        val postcodeRegex =     ("""^((GIR)(0AA)|([A-PR-UWYZ]([0-9]{1,2}|([A-HK-Y][0-9]|[A-HK-Y][0-9]([0-9]|[ABEHMNPRV-Y]))|""" +
+    //          """[0-9][A-HJKS-UW]))([0-9][ABD-HJLNP-UW-Z]{2})|(([A-Z]{1,4})(1ZZ))|((BFPO)([0-9]{1,4})))$""").r
+    //        !postcodeRegex.pattern.matcher(noSpacesPostcode).matches()
+  }
+
+  private def unparsableLocalDate(str: String) = {
+    !"""^\d{8}$""".r.pattern.matcher(str).matches()
+  }
+
+  private def yearFromDate(date: String): Try[Int] = {
```

Comment:
You're not doing anything with the `Left` (the `IllegalArgumentException` is not used). An `Option` seems to be sufficient here